### PR TITLE
utilities: add mass update script

### PR DESCRIPTION
**Pull request by Noki**
This scripts aims to improve mass updates. The script checks for availability, free memory in comparison with the image size and a successful file transfer.

Note: This script will skip some low memory devices and could be improved so it could also flash these in the future, but for now I did not really care.

### CODE REVIEW

File: mass-update.sh [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        ssh-keyscan "$HOSTNAME" >> ~/.ssh/known_hosts
+
+        # Check memory on remote host
+        MEMORY=$(ssh "root@$HOSTNAME" "free | awk 'NR==2 {print \$4}'")
+        if [ "$MEMORY" -ge $(( $(stat -c %s "$FILE_PATH") / 1024 + 3072 )) ]; then  # File size in KB + 3 MB
+            echo "Memory on $HOSTNAME is sufficient ($MEMORY KB)"
+
+            # SCP the file
+            echo "Copying $FILENAME to $HOSTNAME:/tmp/"
+            if scp "$FILE_PATH" "root@$HOSTNAME:/tmp/"; then
```
Comment (pktpls) [inline mass-update.sh:85]: `scp -O` if the destination host doesn't have `openssh-sftp-server` installed

File: mass-update.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                # Wait for hostname to become unreachable
+                echo "Waiting for $HOSTNAME to become unreachable..."
+                while ping -c 1 "$HOSTNAME" >/dev/null 2>&1; do sleep 1; done
+
+                # Wait for 20 seconds before checking hostname reachability again
+                sleep 20
+
+                # Debug output: Waiting for hostname to become reachable again
+                echo "Waiting for $HOSTNAME to become reachable again..."
+                while ! ping -c 1 "$HOSTNAME" >/dev/null 2>&1; do sleep 1; done
```
Comment (pktpls) [inline mass-update.sh:100]: `ping -c` can be combined with `ping -w` so you don't need the loop

File: mass-update.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    echo "Hostname: $HOSTNAME"
+
+    # Check if hostname is reachable
+    echo "Checking if $HOSTNAME is reachable..."
+    if ping -c 1 "$HOSTNAME" >/dev/null 2>&1; then
+        echo "Hostname $HOSTNAME is reachable"
+
+        # Removing old host key
+        echo "Removing old host key for $HOSTNAME"
+        ssh-keygen -R "$HOSTNAME" >/dev/null 2>&1
```
Comment (pktpls) [inline mass-update.sh:72]: You can circumvent all the host key stuff: `sshnocheck='ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no'`

**Comment by pktpls** [comment]
LGTM, small comments, `scp -O` is the only important one

File: mass-update.sh (hunk 2 above)
Comment (Noki) [reply mass-update.sh:92]: The idea of the loop is to be be able to wait for quite some time and to be stuck if the device does not come back. Exiting after a deadline e.g. 120s would of course be better at that point. I don't get how to achieve this with the combination of `-c` and `-w` as `ping -c 1 -w 120 nonexisting.olsr` exits immediately with a "Name or service not known" error.

**Pull request merged**